### PR TITLE
fix #274 Stripper bad attributes on a copy paste in the composer

### LINKED ISSUE #274: Paragraph is collapsed display when pasted in the browser

### PR DESCRIPTION
**Pull request by fabienmoyon**

### CODE REVIEW

File: src/linagora.esn.unifiedinbox/app/components/composer/body-editor/html/composer-body-editor-html.controller.js [hunk 1]
```diff
@@ -60,6 +61,23 @@ angular.module('linagora.esn.unifiedinbox')
       self.onBodyUpdate({ $body: $element.find('.summernote').summernote('code') });
     }
 
+    function onSummernotePaste(event) {
+      event.preventDefault();
+      var text = event.originalEvent.clipboardData.getData('text/plain'),
+        html = event.originalEvent.clipboardData.getData('text/html');
+      var badAttributes = ['line-height'];
```
Comment (oubchid) [inline src/linagora.esn.unifiedinbox/app/components/composer/body-editor/html/composer-body-editor-html.controller.js:68]: `const`

**Review by oubchid — DISMISSED** [review]

**Review by chibenwa — COMMENTED** [review]
No tests ?

File: src/linagora.esn.unifiedinbox/app/components/composer/body-editor/html/composer-body-editor-html.controller.js [hunk 2]
```diff
@@ -60,6 +61,23 @@ angular.module('linagora.esn.unifiedinbox')
...
     }
 
+    function onSummernotePaste(event) {
+      event.preventDefault();
+      var text = event.originalEvent.clipboardData.getData('text/plain'),
+        html = event.originalEvent.clipboardData.getData('text/html');
+      const badAttributes = ['line-height'];
+
+      if (html) {
+        var attributeStripper = new RegExp('(' + badAttributes.join('|') + '\s*:\s*(\w+|\d\.\d+)%?\;?)'); // eslint-disable-line
```
Comment (alagane) [inline src/linagora.esn.unifiedinbox/app/components/composer/body-editor/html/composer-body-editor-html.controller.js:71]: Try it on https://regex101.com/ for example, there is a problem with you "joined group" priority. `(first-attribute)|(second-attribute)|(last-attribute\s*:\s*(\w+|\d\.\d+)%?\;?)` that is how is is interpreted, I think it is not what you wanted?

Can you explain how does this regex work?

**Review by alagane — CHANGES_REQUESTED** [review]

File: src/linagora.esn.unifiedinbox/app/components/composer/body-editor/html/composer-body-editor-html.controller.js (hunk 2 above)
Comment (fabienmoyon) [reply src/linagora.esn.unifiedinbox/app/components/composer/body-editor/html/composer-body-editor-html.controller.js:71]: https://regex101.com/r/cKAdRC/1/

**Pull request closed without merging**